Feature/soql where clause support (#2663)
SFDO-Tooling/CumulusCI: docs/data.rst

            CustomCheckbox__c: True
            CustomDateField__c: 2019-01-01

> The ``soql_filter`` key allows to specify WHERE clause that should be used when extracting data from salesforce org::

Reviewer: ```suggestion
The ``soql_filter`` key allows to specify a WHERE clause that should be used when extracting data from your Salesforce org:
```

        Account:
          sf_object: Account
          table: Account
          fields:
            - Name
            - Industry
            - Type
          soql_filter: "Industry = 'Higher Education' OR Type = 'Higher Education'"

> Note that trying to load data that is extracted using ``soql_filter`` may cause "invalid cross reference id" error if related object records are missing in local database. Use this feature only if you fully understand how `CumulusCI load data task <https://cumulusci.readthedocs.io/en/stable/data.html#load-dataset>`_ resolves references to related records when loading data to Salesforce Org.

Reviewer: ```suggestion
Note that trying to load data that is extracted using ``soql_filter`` may cause "invalid cross reference id" errors if related object records are filtered on extract. Use this feature only if you fully understand how `CumulusCI load data task <https://cumulusci.readthedocs.io/en/stable/data.html#load-dataset>`_ resolves references to related records when loading data to a Salesforce org.
```
